oak-core/src/test/java/org/apache/jackrabbit/oak/plugins/nodetype/TypeEditorTest.java

```diff
@@ -324,4 +326,34 @@ public void malformedUUID() throws CommitFailedException {
             assertTrue(e.isConstraintViolation());
         }
     }
+
+    @Test
+    public void addChildNodeNotStrictDefaultPrimaryTypeIfUnspecified() throws CommitFailedException {
+        // set strict to 'false' to allow unset jcr:primaryType on new node
+        EditorHook hook = new EditorHook(new TypeEditorProvider(false));
+
+        NodeState root = INITIAL_CONTENT;
+        NodeBuilder builder = root.builder();
+
+        NodeBuilder policyBuilder = builder.child("conf")
+                .setProperty(JCR_PRIMARYTYPE, "nt:folder", Type.NAME)
+                .setProperty(JCR_MIXINTYPES, ImmutableList.of(AccessControlConstants.MIX_REP_ACCESS_CONTROLLABLE), Type.NAMES)
+                .child("rep:policy")
+                // create the policy node with the correct primary type in the before state
+                .setProperty(JCR_PRIMARYTYPE, AccessControlConstants.NT_REP_ACL, Type.NAME);
+
+        NodeState before = builder.getNodeState();
+        policyBuilder.child("allow")
```

Comment:
please add a comment here as well: because the crucial piece is that this 'allow' child has no jcr:primaryType property and with 'strict' false it should be set to the default primary type defined by the parent.
so, it's the combination of the parent having a primary type that defines a default-primary-type for child nodes and the fact that this 'allow' child lacks the primary type. and extra comment here would make the test a lot easier to read.... (i had to go back to your bug report to understand it)